test/ng/compileSpec.js

```diff
@@ -3756,6 +3760,7 @@ describe('$compile', function() {
         module('my');
         inject(function($compile, $rootScope) {
           element = $compile('<c1 prop1="val"></c1>')($rootScope);
+          $rootScope.$apply();
```

Comment:
If we have no digest before changing the value of `val` then effectively the system thinks that `val` has not changed, so `$onChanges` is not called.